### PR TITLE
Subpage UI changes

### PR DESCRIPTION
**Pull request by tbeidlershenk**
### Changes
1. Rebuilt the UI of posts in ```ListPage```
2. Rebuilt the UI of ```PostPage``` when a user clicks on a map item or page in the list view
3. Rebuilt the UI of ```ItemRequest``` and renamed to ```SubmissionPage```

### Notes
Can you let me know if you like the updated pages?
@jeffzheng13 when we merge PRs from now on can we choose "Squash and merge" so our commit history on main is more condensed?
and also we should get in the habit of rebasing our branches on main before submitting PRs. it removes the possibility of merge conflict

### CODE REVIEW

File: lib/services/firestore_service.dart [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
+import 'package:lost_found_steelhacks/authentication/user.dart';
 import 'package:lost_found_steelhacks/data/item.dart';
 import 'package:rxdart/rxdart.dart';
 
 class FirestoreService {
   //create reference to database
   final lostRef = FirebaseFirestore.instance.collection('lost');
-
   final foundRef = FirebaseFirestore.instance.collection('found');
+  final userRef = FirebaseFirestore.instance.collection('users');
```
Comment (jeffzheng13) [inline lib/services/firestore_service.dart:10]: I don't see this being used anywhere. Do we still need this line?

File: lib/cards/post_card.dart [hunk 2]
```diff
@@ -14,9 +14,69 @@ class PostCard extends StatelessWidget {
...
+      return "Yesterday";
+    }
+    return "${date.month}-${date.day}";
+  }
+
   @override
   Widget build(BuildContext context) {
     final theme = Theme.of(context).extension<AppTheme>()!;
+    final BoxDecoration bodyDecoration = BoxDecoration(
+      color: theme.medium,
```
Comment (jeffzheng13) [inline lib/cards/post_card.dart:33]: White text is difficult to read on the green background, could we try black text?

File: lib/pages/submission_page.dart [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+              child: Text("Submit a found item", style: theme.darkTitleStyle),
+            ),
+            IconButton(
+                color: theme.dark,
+                icon: const Icon(Icons.keyboard_return_rounded),
+                onPressed: () => routeBack(context))
+          ],
+        );
+
+    Widget buildForm() => FastForm(
```
Comment (jeffzheng13) [inline lib/pages/submission_page.dart:141]: I would still like to be able to see what image I uploaded. Feels odd to submit without being able to see image selected.

File: lib/pages/submission_page.dart [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+                icon: const Icon(Icons.upload),
+                label: Text(bytes == null ? "Upload image" : "Change image"),
+                onPressed: () => loadImageIntoMemory()),
+          ),
+        );
+
+    Widget buildSubmitButton() => loading
+        ? const Loading()
+        : IconButton(
+            icon: const Icon(Icons.add_box_rounded),
```
Comment (jeffzheng13) [inline lib/pages/submission_page.dart:206]: This icon isn't really explanatory. Could we keep the button with the word 'Submit' so it clearly indicated as such?

File: lib/pages/submission_page.dart [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+    Widget buildUploadButton() => Padding(
+          padding: const EdgeInsets.all(8.0),
+          child: Container(
+            decoration: BoxDecoration(
+                color: theme.light,
+                border: Border.all(color: theme.medium, width: 3),
+                borderRadius: const BorderRadius.all(Radius.circular(3))),
+            child: TextButton.icon(
+                icon: const Icon(Icons.upload),
+                label: Text(bytes == null ? "Upload image" : "Change image"),
```
Comment (jeffzheng13) [inline lib/pages/submission_page.dart:198]: Label text is purple for some reason.

File: lib/pages/submission_page.dart [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+                  items: itemNameOptions,
+                  onChanged: (value) => setState(() => itemName = value!),
+                ),
+              ),
+              Padding(
+                padding: const EdgeInsets.all(8.0),
+                child: FastTextField(
+                  maxLines: 5,
+                  name: "description_text_field",
+                  labelText: "Description",
```
Comment (jeffzheng13) [inline lib/pages/submission_page.dart:177]: Can we make description labelText smaller? It's so jarring 😅

File: lib/pages/submission_page.dart [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+    Widget buildHeader() => Row(
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+          children: [
+            Padding(
+              padding: const EdgeInsets.all(8.0),
+              child: Text("Submit a found item", style: theme.darkTitleStyle),
+            ),
+            IconButton(
+                color: theme.dark,
+                icon: const Icon(Icons.keyboard_return_rounded),
```
Comment (jeffzheng13) [inline lib/pages/submission_page.dart:136]: Are we sure we want this as our exit icon? I prefer the 'X' to be honest. It's much clearer in my opinion, but we can discuss.

File: lib/pages/post_page.dart [hunk 8]
```diff
@@ -1,142 +1,108 @@
...
+              icon: Icon(Icons.chat),
+              onPressed: () => routeHome(2, context)),
+          Spacer(),
+          Padding(
+            padding: const EdgeInsets.all(5.0),
+            child: Text(getDate(), style: theme.darkRegularStyle),
+          ),
+          Padding(
+              padding: const EdgeInsets.all(5.0),
+              child: Icon(Icons.access_time_sharp, color: theme.dark)),
```
Comment (jeffzheng13) [inline lib/pages/post_page.dart:75]: Can we swap date and clock icon order?

**Comment by jeffzheng13** [comment]
Also can you add to after line 26 of list_page.dart:
`   objects.sort((a, b) => b.timeCreated.compareTo(a.timeCreated));`
so the items are sorted by most recent date first.

**Comment by jeffzheng13** [comment]
Did not test adding markers to map, however it looks functional and connected to backend. Small nitpicks and improvements above, but looking great and good work @tbeidlershenk!

File: lib/services/firestore_service.dart (hunk 1 above)
Comment (tbeidlershenk) [reply lib/services/firestore_service.dart:10]: yep I'll remove

File: lib/pages/submission_page.dart (hunk 6 above)
Comment (tbeidlershenk) [reply lib/pages/submission_page.dart:177]: image
Is this preferable?

Comment (tbeidlershenk) [reply lib/pages/submission_page.dart:177]: I'm not able to make the reminder text in the box itself smaller with this widget.

File: lib/pages/submission_page.dart (hunk 7 above)
Comment (tbeidlershenk) [reply lib/pages/submission_page.dart:136]: TBH I prefer the arrow but I don't mind moving back to the X

File: lib/pages/post_page.dart (hunk 8 above)
Comment (tbeidlershenk) [reply lib/pages/post_page.dart:75]: I've tried both, I think it looks better how it currently is. But here's the other: 
image
Icons between posts don't line up nicely if we put it on the left

**Comment by jeffzheng13** [comment]
Added comments and a few follow ups but once that's in there we can merge and have a new revamped UI. 👏

File: lib/pages/post_page.dart [hunk 9]
```diff
@@ -1,142 +1,108 @@
...
-                            crossAxisAlignment: CrossAxisAlignment.start,
-                            children: [
-                              RichText(
-                                  textAlign: TextAlign.start,
-                                  text: TextSpan(children: [
-                                    // Text should pull global font
-                                    TextSpan(
-                                        text: "Description\n",
-                                        style: theme.subtitleStyle),
+    Widget buildBody() => Padding(
```
[the remark this answers is not in the document]
Comment (tbeidlershenk) [reply lib/pages/post_page.dart:56]: I think its not worth it to implement something we won't end up using

File: lib/pages/post_page.dart [hunk 10]
```diff
@@ -1,142 +1,108 @@
...
+  String getDate() {
+    DateTime now = DateTime.now();
+    DateTime date = item.timeCreated.toDate();
+    if (now.difference(date).inDays == 0) {
+      return "Today";
+    }
+    if (now.difference(date).inDays == 1) {
+      return "Yesterday";
+    }
+    return "${date.month}-${date.day}";
```
[the remark this answers is not in the document]
Comment (tbeidlershenk) [reply lib/pages/post_page.dart:26]: Yep!!

**Review by jeffzheng13 — APPROVED** [review]
Reviewed + Satisfied!

**Pull request merged**